src/detect-tcpmss.c

```diff
@@ -62,6 +62,25 @@ void DetectTcpmssRegister(void)
     return;
 }
 
+/**
+ * \brief Return human readable value for tcp.mss mode
+ *
+ * \param mode uint8_t DetectU16Data tcp.mss mode value
+ */
+const char *TcpmssModeToString(uint8_t mode)
+{
+    switch (mode) {
+        case 1:
+            return "less than";
+        case 3:
+            return "greater than";
+        case 5:
+            return "range";
+        default:
+            return NULL;
```

Comment:
> As we've discussed in the SV PR, there are more possible values for `mode`. Check
> https://github.com/OISF/suricata/blob/master/src/detect-engine-uint.h#L30-L38 for all possible values. It seems that there is no check that would indicate some of those are not possible, so I'm assuming Suri should be able to parse and process any for tcp.mss
> 
> Aside from that,  do notice that `DetectU16Data.mode` is of type `DetectUintMode`, so we should use a parameter of that type, this is more secure, as it restricts the possible values for mode.
> 
> This also makes the code itself more human readable, and avoids magic numbers :)
> 
> 
> 

I explained previously, in the userguide instructions for `tcp.mss`, the formats for rules only includes Less than, Greater than, range and equal.

Please can you elaborate on your second point? I don't quite understand; help with an instance.

Reply:
> Thinking that maybe default would better be "" or `equal` instead of `NULL`...

I thought of this too.

I'll do that.

Reply:
> > As we've discussed in the SV PR, there are more possible values for `mode`. Check
> > https://github.com/OISF/suricata/blob/master/src/detect-engine-uint.h#L30-L38 for all possible values. It seems that there is no check that would indicate some of those are not possible, so I'm assuming Suri should be able to parse and process any for tcp.mss
> > Aside from that,  do notice that `DetectU16Data.mode` is of type `DetectUintMode`, so we should use a parameter of that type, this is more secure, as it restricts the possible values for mode.
> > This also makes the code itself more human readable, and avoids magic numbers :)
> 
> I explained previously, in the userguide instructions for `tcp.mss`, the formats for rules only includes Less than, Greater than, range and equal.

I see. Sorry, I missed that point among the threads.
Even if the userguide doesn't specify it, I think it's best if we handle those values somehow, because the data type that stores them could receive those, which means Suricata can parse them if a rule is written with them, and from the code I couldn't see a place where we reject such values, so I think it's best if we treat/handle them.

> 
> Please can you elaborate on your second point? I don't quite understand; help with an instance.

The second point is about the datatype for the struct field `mode` that we are matching on here. While in `DetectIpOpts.code` the field is a `uint16_t`, for `DetectUintData_u16.mode`, the type is `DetectUintMode` so it is best if the function parameter is that, instead of a `uint8_t`, as the DetectUintMode will have a very specific set of possible values, making the whole function easier to treat - and also allowing you to use something like
```c
        case DetectUintModeLt:
            return "less than";
```
instead of `case 1` for instance.

So, the parameter here should be of type `DetectUintMode` instead of `uint8_t`. And then you can check the possible types on that file that I shared before. Does this help?

Reply:
> > > As we've discussed in the SV PR, there are more possible values for `mode`. Check
> > > https://github.com/OISF/suricata/blob/master/src/detect-engine-uint.h#L30-L38 for all possible values. It seems that there is no check that would indicate some of those are not possible, so I'm assuming Suri should be able to parse and process any for tcp.mss
> > > Aside from that,  do notice that `DetectU16Data.mode` is of type `DetectUintMode`, so we should use a parameter of that type, this is more secure, as it restricts the possible values for mode.
> > > This also makes the code itself more human readable, and avoids magic numbers :)
> > 
> > I explained previously, in the userguide instructions for `tcp.mss`, the formats for rules only includes Less than, Greater than, range and equal.
> 
> I see. Sorry, I missed that point among the threads.
> Even if the userguide doesn't specify it, I think it's best if we handle those values somehow, because the data type that stores them could receive those, which means Suricata can parse them if a rule is written with them, and from the code I couldn't see a place where we reject such values, so I think it's best if we treat/handle them.
> 
> > 
> > Please can you elaborate on your second point? I don't quite understand; help with an instance.
> 
> The second point is about the datatype for the struct field `mode` that we are matching on here. While in `DetectIpOpts.code` the field is a `uint16_t`, for `DetectUintData_u16.mode`, the type is `DetectUintMode` so it is best if the function parameter is that, instead of a `uint8_t`, as the DetectUintMode will have a very specific set of possible values, making the whole function easier to treat - and also allowing you to use something like
> ```c
>         case DetectUintModeLt:
>             return "less than";
> ```
> instead of `case 1` for instance.
> 
> So, the parameter here should be of type `DetectUintMode` instead of `uint8_t`. And then you can check the possible types on that file that I shared before. Does this help?

OK! I understand you.

Thank you for the feedback.